src/pages/index.js

```diff
@@ -103,70 +116,176 @@ const settings = {
+    .then((res) => {
+      console.log(res);
+      createCard(res);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
 addCardPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#edit-modal",
-  handleProfileFormSubmit
-);
+const editProfilePopup = new PopupWithForm("#edit-modal", (card) => {
+  editProfilePopup.setButtonText(true);
+  return api
+    .editUserInfo(card)
+    .then((res) => {
+      console.log(res);
+      userInfo.setUserInfo(res.name, res.about);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
 editProfilePopup.setEventListeners();
 
-//function renderCard(cardData, wrapper) {
-//const newCard = new Card(cardData, cardSelector, handleCardClick);
-//cardSection.addItem(newCard.getView());
-//}
+avatarModalPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  cardListSelector
-);
+const editFormValidator = new FormValidator(settings, editFormElement);
+const addFormValidator = new FormValidator(settings, addFormElement);
 
-cardSection.renderItems();
+const editAvatarValidator = new FormValidator(settings, avatarModalElement);
+
+editFormValidator.enableValidation();
+addFormValidator.enableValidation();
+editAvatarValidator.enableValidation();
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setAvatar(userData.avatar);
+    console.log(cards);
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: createCard,
+      },
+      cardListSelector
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 const imagePreviewPopup = new PopupWithImage("#card-image");
 imagePreviewPopup.setEventListeners();
 
-function handleCardClick(name, link) {
-  imagePreviewPopup.open(name, link);
-}
-
-function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo(data.title, data.description);
-  editProfilePopup.close();
-}
+const deleteCardModal = new PopupWithConfirmation("#delete-modal", (card) => {
+  console.log("card", card);
+});
+deleteCardModal.setEventListeners();
 
 function createCard(cardData) {
-  const newCard = new Card(cardData, cardSelector, handleCardClick);
+  const newCard = new Card(
+    cardData,
+    cardSelector,
+    () => {
+      imagePreviewPopup.open(cardData.name, cardData.link);
+    },
+    (card) => {
+      deleteCardModal.open(cardData);
+      deleteCardModal.setAction(() => {
+        deleteCardModal.setButtonText(true);
+        api
+          .deleteCard(card._id)
+          .then(() => {
+            console.log(card);
+            card.deleteCard();
+            deleteCardModal.close();
+          })
+          .catch((err) => {
+            console.error(err);
+          })
+          .finally(() => {
+            deleteCardModal.setButtonText(false);
```

Comment:
Good job that you return the button text in `finally`

Reply:
Thank you!